lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function indexOf(searchElement, fromIndex) {
+
+    var k;
+
+    if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+    }
+
+    var o = this;
```

Comment:
Yes it does not work, it throws a TypeError. The spec does not precise if it should work for a number (for example or any object different from array or string). Under Google Chrome, the behavior is the same : it throws a TypeError. I'm not sure that we really should do this check.

---
In reply to: [111824987](https://github.com/Microsoft/ChakraCore/pull/2845#discussion_r111824987) [](ancestors = 111824987)